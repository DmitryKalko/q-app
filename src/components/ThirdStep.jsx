import React from "react";
import Radio from "@mui/material/Radio";
import RadioGroup from "@mui/material/RadioGroup";
import FormControlLabel from "@mui/material/FormControlLabel";
import FormControl from "@mui/material/FormControl";

import data from "../store/data";
import { observer } from "mobx-react-lite";

const ThirdStep = observer(({ toFinishStep, toPrevStep, lightTheme }) => {
  const setNotifications = (e) => {
    data.setNotifications(e.target.value);
  };

  return (
    <div className="third-step">
      <FormControl>
        <RadioGroup
          aria-labelledby="demo-radio-buttons-group-label"
          defaultValue=""
          value={data.store.notifications}
          name="radio-buttons-group"
          sx={lightTheme ? { color: "white" } : { color: "#6d6d6d" }}
          onChange={setNotifications}
        >
          <FormControlLabel value="day" control={<Radio />} label="Every day" />
          <FormControlLabel
            value="week"
            control={<Radio />}
            label="Once a week"
          />
          <FormControlLabel
            value="month"
            control={<Radio />}
            label="Once a month"
          />
          <FormControlLabel
            value="no"
            control={<Radio />}
            label="I don't need notifications"
          />
        </RadioGroup>
      </FormControl>
      <div className="btns-block">
        <button
          className="btn"
          onClick={toPrevStep}
          style={lightTheme ? { color: "white" } : { color: "#6d6d6d" }}
        >
          BACK
        </button>
        <button
          className="btn"
          onClick={toFinishStep}
          style={lightTheme ? { color: "white" } : { color: "#6d6d6d" }}
        >
          NEXT
        </button>
      </div>
    </div>
  );
});
export default ThirdStep;
