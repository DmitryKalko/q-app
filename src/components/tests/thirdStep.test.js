import { shallow } from "enzyme";
import React from "react";
import ThirdStep from "../ThirdStep";


const setUp = () => shallow(<ThirdStep  />);

describe('ThirdStep component', () => {
    let component = setUp();

    test('should render ThirdStep component', () => {
        expect(component).toMatchSnapshot();
    });


    it('ThirdStep inputs', () => {
        const testState = { notification: 'day' };
        expect(component.find('#radio-group').prop('value')).toEqual(null);
        component.find('#radio-group').simulate('change', { target: { notification: 'day' } });
        expect(testState.notification).toEqual('day');
      });
})