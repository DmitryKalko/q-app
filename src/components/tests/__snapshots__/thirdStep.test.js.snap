// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThirdStep component should render ThirdStep component 1`] = `
<div
  className="third-step"
>
  <ForwardRef(FormControl)>
    <ForwardRef(RadioGroup)
      aria-labelledby="demo-radio-buttons-group-label"
      defaultValue=""
      id="radio-group"
      name="radio-buttons-group"
      onChange={[Function]}
      sx={
        Object {
          "color": "#6d6d6d",
        }
      }
      value={null}
    >
      <ForwardRef(FormControlLabel)
        control={<ForwardRef(Radio) />}
        label="Every day"
        value="day"
      />
      <ForwardRef(FormControlLabel)
        control={<ForwardRef(Radio) />}
        label="Once a week"
        value="week"
      />
      <ForwardRef(FormControlLabel)
        control={<ForwardRef(Radio) />}
        label="Once a month"
        value="month"
      />
      <ForwardRef(FormControlLabel)
        control={<ForwardRef(Radio) />}
        label="I don't need notifications"
        value="no"
      />
    </ForwardRef(RadioGroup)>
  </ForwardRef(FormControl)>
  <div
    className="btns-block"
  >
    <button
      className="btn"
      style={
        Object {
          "color": "#6d6d6d",
        }
      }
    >
      BACK
    </button>
    <button
      className="btn"
      style={
        Object {
          "color": "#6d6d6d",
        }
      }
    >
      NEXT
    </button>
  </div>
</div>
`;
